diff.c

```diff
@@ -3900,6 +3901,13 @@ static int reuse_worktree_file(struct index_state *istate,
 	if (!FAST_WORKING_DIRECTORY && !want_file && has_object_pack(oid))
 		return 0;
 
+	/*
+	 * If this path does not match our sparse-checkout definition,
+	 * then the file will not be in the working directory.
+	 */
+	if (!path_in_sparse_checkout(name, istate))
```

Comment:
In https://github.com/microsoft/git/pull/410/files#diff-f89b08f027f583c8719c7a22dba3ccc888b98aafc084c40df006c4e82a0ca278R755-R759, you had to add more conditionals and had to ensure that the patterns were initialized. Is that not necessary here?

Reply:
`path_in_sparse_checkout()` calls `init_sparse_checkout_patterns()` so we don't need to do that any more!